src/sidebar/test/service-url-test.js

```diff
@@ -1,20 +1,166 @@
 'use strict';
 
-var serviceUrl = require('../service-url');
+var proxyquire = require('proxyquire');
 
-describe('serviceUrl', function () {
-  var service;
+/** Return a fake annotationUI object. */
+function fakeAnnotationUI() {
+  var links = null;
+  return {
+    updateLinks: function(newLinks) {
+      links = newLinks;
+    },
+    getState: function() {
+      return {links: links};
+    },
+  };
```

Comment:
This `annotationUI` fake is interesting. From `serviceUrl`'s point of view, the contract of `annotationUI` is just:

* You can set the links value (and it will do the magic re-rendering of everything)
* You can get the links value, and it will return the same object that was set
* Initially, before set, get returns null. Or it could be undefined

This is actually the contract of a standard JavaScript object:

```javascript
> var annotationUI = {}
> annotationUI.links
undefined
> annotationUI.links = 'foo'
> annotationUI.links
"foo"
```

I wonder if `annotationUI` could use getters and setters so that it presents the interface of a normal JavaScript object to `serviceUrl`? That would simplify `serviceUrl`'s code, but also in the tests `fakeAnnotationUI` would just be: `var fakeAnnotationUI = {};`.